gc/gc-base/pom.xml

```diff
@@ -41,6 +41,12 @@
       <version>${project.version}</version>
       <scope>provided</scope>
     </dependency>
+    <!--- We cannot use version > 0.43 beacuse it uses a kryo version 5.x that is incompatible with out spark (requires 4.x) -->
```

Comment:
it provides needed  `ImmutableMapSerializer` and `UnmodifiableCollectionsSerializer`